server/README.md

```diff
@@ -0,0 +1,149 @@
+# Flask API README
+
+This is a Flask API for user registration, login, and authentication. It includes user registration, login, authentication, and some protected routes. This README provides an overview of the application and instructions on how to run it using Docker.
+
+## Table of Contents
+
+- [Application Overview](#application-overview)
+- [Requirements](#requirements)
+- [Installation](#installation)
+- [Running the Application](#running-the-application)
+- [API Endpoints](#api-endpoints)
+- [Docker](#docker)
+
+## Application Overview
+
+### Dependencies
+
+This Flask application utilizes the following dependencies:
+
+- Flask for building the API.
+- Flask-CORS for Cross-Origin Resource Sharing configuration.
+- Flask-SQLAlchemy for database interaction.
+- Flask-Login for user session management.
+- Passlib for password hashing.
+- Python-dotenv for environment variable management.
+
+### Functionality
+
+This Flask API provides the following functionality:
+
+- User registration with username, password, email, and user type (candidate or company).
+- User authentication and login.
+- User session management using Flask-Login.
+- Authentication middleware for protecting routes.
+- User data stored in a PostgreSQL database.
+
+## Requirements
+
+Before you begin, ensure you have the following requirements installed on your system:
+
+- Python 3.11 or higher
+- Docker (optional, for Docker deployment)
+
+## Installation
+
+1. Clone this repository to your local machine:
+
+   ```bash
+   git clone https://github.com/your-username/your-repo.git
+   ```
+
+2. Navigate to the project directory:
+
+```bash
+cd your-repo
+```
+
+3. Create a virtual environment and activate it (optional but recommended):
+
+```bash
+python -m venv venv
+source venv/bin/activate
+```
+
+4. Install the required Python packages from the requirements.txt file:
+
+```bash
+pip install -r requirements.txt
+```
+
+5. Create a .env file in the project root and define the following environment variables:
+
+- **DATABASE_URI_TEST:** The URI for your PostgreSQL database.
+- **SECRET_KEY:** A secret key for session management and security.
+
+Example .env file:
+
+```plaintext
+DATABASE_URI_TEST=postgresql://your-db-user:your-db-password@localhost/your-db-name
+SECRET_KEY=your-secret-key
+```
+
+## Running the Application
+
+To run the Flask application locally, use the following command:
+
+```bash
+python3 main.py
+```
+
+The application will be accessible at http://localhost:5001.
+
+## API Endpoints
```

Comment:
Don't need to add this here, we might want to create a separate document (google docs) with all the details.